worlds/oot_soh/location_access/overworld/market.py

```diff
@@ -5,8 +5,273 @@
+    connect_regions(Regions.MARKET_GUARD_HOUSE, world, [
+        (Regions.MARKET_ENTRANCE, lambda bundle: True)
+    ])
+    
+    ## Market Bazaar
+    # Locations
+    add_locations(Regions.MARKET_BAZAAR, world, [
+        (Locations.MARKET_BAZAAR_ITEM1, lambda bundle: True),
+        (Locations.MARKET_BAZAAR_ITEM2, lambda bundle: True),
+        (Locations.MARKET_BAZAAR_ITEM3, lambda bundle: True),
+        (Locations.MARKET_BAZAAR_ITEM4, lambda bundle: True),
+        (Locations.MARKET_BAZAAR_ITEM5, lambda bundle: True),
+        (Locations.MARKET_BAZAAR_ITEM6, lambda bundle: True),
+        (Locations.MARKET_BAZAAR_ITEM7, lambda bundle: True),
+        (Locations.MARKET_BAZAAR_ITEM8, lambda bundle: True),
+    ])
+    # Connections
+    connect_regions(Regions.MARKET_BAZAAR, world, [
+        (Regions.MARKET, lambda bundle: True)
+    ])
+    
+    ## Market Mask Shop
+    # Events
+    add_events(Regions.MARKET_MASK_SHOP, world, [
+        (EventLocations.MARKET_MASK_SHOP_ALL_MASKS, Events.CAN_BORROW_MASKS, lambda bundle: (has_item(Items.ZELDAS_LETTER, bundle) and
+                                                                                   (has_item(Events.KAKARIKO_GATE_OPEN, bundle) or world.options.kakariko_gate.value == 1) or
+                                                                                   world.options.complete_mask_quest.value == 1)),
+        (EventLocations.MARKET_MASK_SHOP_SKULL_MASK, Events.CAN_BORROW_SKULL_MASK, lambda bundle: (world.options.complete_mask_quest.value or
+                                                                                        has_item(Events.CAN_BORROW_MASKS, bundle))),
+        (EventLocations.MARKET_MASK_SHOP_SPOOKY_MASK, Events.CAN_BORROW_SPOOKY_MASK, lambda bundle: (world.options.complete_mask_quest.value or
+                                                                                        has_item(Events.CAN_BORROW_MASKS, bundle))),
+        (EventLocations.MARKET_MASK_SHOP_BUNNY_HOOD, Events.CAN_BORROW_BUNNY_HOOD, lambda bundle: (world.options.complete_mask_quest.value or
+                                                                                        has_item(Events.CAN_BORROW_MASKS, bundle)))
+    ])
+    # Connections
+    connect_regions(Regions.MARKET_MASK_SHOP, world, [
+        (Regions.MARKET, lambda bundle: True)
+    ])
+    
+    ## Market Shooting Gallery
+    # Locations
+    add_locations(Regions.MARKET_SHOOTING_GALLERY, world, [
+        (Locations.MARKET_SHOOTING_GALLERY, lambda bundle: (is_child(bundle) and has_item(Items.CHILD_WALLET, bundle)))
+    ])
+    # Connections
+    connect_regions(Regions.MARKET_SHOOTING_GALLERY, world, [
+        (Regions.MARKET, lambda bundle: True)
+    ])
+    
+    ## Market Bombchu Bowling
+    # Events
+    add_events(Regions.MARKET_BOMBCHU_BOWLING, world, [
+        (EventLocations.MARKET_BOMBCHU_BOWLING, LocalEvents.CAN_PLAY_BOWLING, lambda bundle: (has_item(Items.CHILD_WALLET, bundle)))
+    ])
```

Comment:
The event location needs a unique name to the regular location. `MARKET_BOMBCHU_BOWLING_GAME` or something?